### PR TITLE
fix: distribute files to machines for external contributors.

### LINKED ISSUE #22308: CI: External contributions are not distributed properly causing long and flaky builds
### Current behavior

Tests are doubled up on CI for external contributors, see [build](https://app.circleci.com/pipelines/github/cypress-io/cypress/39476/workflows/5c2c9690-bd4c-4271-9343-3ef759db4891/jobs/1613218). We ran into this problem pre-10.0 but did't apply the same test distribution rules to our new 10.0 tests https://github.com/cypress-io/cypress/pull/21305. This can lead to long and flaky builds for external contributors.

That job isn't the only one that is duplicating tests, so an audit of the [this build](https://app.circleci.com/pipelines/github/cypress-io/cypress/39476/workflows/5c2c9690-bd4c-4271-9343-3ef759db4891) should be made to find all the jobs that are not distributing properly.

Screen Shot 2022-06-14 at 9 23 12 AM

### Desired behavior

CI should distribute tests properly for 10.0 builds

### Test code to reproduce

Check out an external contributors CI build

### Cypress Version

10.1.0

### Other

_No response_

### PR DESCRIPTION
**Pull request by sainthkh**
- Closes #22308

### User facing changelog

It's not a feature. It's a change for external contributors. Distribute files to machines for faster and less-flaky CI experience. 

### Additional details
- Why was this change necessary? => Tests are run several times and it causes flaky failures.
- What is affected by this change? => N/A
- Any implementation details to explain? => It's the same job I did with #21305.

### Steps to test

Check [this CI result](https://app.circleci.com/pipelines/github/cypress-io/cypress/39483/workflows/4eba9459-15a1-4715-a643-c8acf874359f/jobs/1615431).

### How has the user experience changed?

N/A

### PR Tasks

- [na] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by sainthkh** [comment]
Now, CI distributes test files to machines. 

I intentionally left `run-reporter-component` failed, because it has only 2 test files but has 7 machines. I think it's a bug in the CI setting, not the CI script.

File: circle.yml [hunk 1]
```diff
@@ -516,14 +516,45 @@ commands:
...
-            cmd=$([[ <<parameters.percy>> == 'true' ]] && echo 'yarn percy exec --parallel -- --') || true
-            DEBUG=<<parameters.debug>> \
-            CYPRESS_KONFIG_ENV=production \
-            CYPRESS_RECORD_KEY=${TEST_LAUNCHPAD_RECORD_KEY:-$MAIN_RECORD_KEY} \
-            PERCY_PARALLEL_NONCE=$CIRCLE_SHA1 \
-            PERCY_ENABLE=${PERCY_TOKEN:-0} \
-            PERCY_PARALLEL_TOTAL=-1 \
-            $cmd yarn workspace @packages/<<parameters.package>> cypress:run:<<parameters.type>> --browser <<parameters.browser>> --record --parallel --group <<parameters.package>>-<<parameters.type>>
+            echo Current working directory is $PWD
+            echo Total containers $CIRCLE_NODE_TOTAL
```
[the remark this answers is not in the document]
Comment (sainthkh) [reply circle.yml]: I named it `print-ci-settings`. The purpose of those lines is to see what's going on in CI for debugging.

[the remark this answers is not in the document]
Comment (sainthkh) [reply circle.yml:520]: I tried refactoring them but it seems that it's not a good idea. My tries can be found at 15c36065cab1d7dd3fa6213144ca442509cbca53.

`run-new-ui-tests` and `run-driver-integration-tests` are really different. They call `cypress:run` in different ways with different options. In addition, `driver` has 'experimentalSessionAndOrigin` parameter while `new-ui` does not. 

Only things they have in common are `$PWD message` and `TESTFILES` variable. I tried to gather them and create a reusable step. But as it uses `TESTFILES` environment variable, it is not persistent between steps. ([I tried `$BASH_ENV` method written here, but it seems that it creates a racing condition.](https://circleci.com/docs/2.0/env-vars#example-configuration-of-environment-variables))

And current `run-new-ui-tests` is a reusable step for new packages. (It is used in app, launchpad, etc.) When session and origin experiment is done and `driver` package is refactored to be `esm`, I guess we can safely use `run-new-ui-tests` with `driver` then.

**Review by flotwig — APPROVED** [review]

**Review by ryanthemanuel — APPROVED** [review]

**Comment by lmiller1990** [comment]
The CI failure was happening locally, too. I tried pushing a commit to get it over the line: https://github.com/cypress-io/cypress/pull/22326/commits/42dbe5497004dc02d6c9b99904fce90014c2379d

**Pull request merged**